int32 encLeft
int32 encRight
